### ISSUE TITLE
descriptor for crystal

### ISSUE THREAD

**Issue by StevenAlice93** [SINGROUP/dscribe#46]
Hello, Thank you for this excellent project! I want to know whether the code can compute discriptors for crystal?

I'm looking forward to your reply.
 
Best wishes!

**Comment by lauri-codes** [comment]
Hi @StevenAlice93,

Sure, there are several options. I would encourage you to read our Open Access article: https://doi.org/10.1016/j.cpc.2019.106949

In short: MBTR, sine matrix, Ewald sum matrix work natively for crystals, ACSF and SOAP also work with a few tricks.

**Comment by StevenAlice93** [comment]
Well done! Thank you! I want to know how to use it to transform the POSCAR into descriptors, and I don't find the API to do that.

**Comment by lauri-codes** [comment]
You can use [ASE](https://wiki.fysik.dtu.dk/ase/) for reading structures from POSCAR files. The resulting `Atoms` object can then be used directly by `dscribe`.

**Comment by StevenAlice93** [comment]
Dear lauri-codes,

  Oh, thank you! I got it. 

  I have a another question: if I have two moleculars CH4 and H20, and I use the ACSF Class to get descriptors for them, I will get a numpy array with shape (3, 36) for H20 and a numpy array with shape (5,36) for CH4. For many machine learning models, they only accept these samples with the same length descriptors, so how do I input these descriptors input a machine learning model?  

  Thank you for your reply!

Best wishes!

**Comment by lauri-codes** [comment]
Hi @StevenAlice93,

I have just added [a basic tutorial](https://singroup.github.io/dscribe/tutorials/basics.html) that should cover your questions.

**Comment by zhanzghang** [comment]
Dear Lauri,


I would like to use SOAP to train some data, it looks like the file with your dataset 'formation_energy.extxyz'. But i have a question in your file, the atoms coordinate come from where(I konw your file from OQMD)? The primitive cell.cif from materials project has atoms coordinate, i don't know these coordinate is right?&nbsp;


Thanks much for your time!





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Lauri Himanen"<notifications@github.com&gt;;
发送时间:&nbsp;2020年4月21日(星期二) 凌晨2:01
收件人:&nbsp;"SINGROUP/dscribe"<dscribe@noreply.github.com&gt;;
抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;;
主题:&nbsp;Re: [SINGROUP/dscribe] descriptor for crystal (#46)





 
Closing for now.

**Comment by lauri-codes** [comment]
Hi @zhanzghang,

The contents of formation_energy.extxyz come from OQMD 1.1. I created those files by using the SQL database that can be downloaded from http://oqmd.org.

**Comment by zhanzghang** [comment]
Dear Lauri,

when I used the Coulomb Matrix and Sine matrix to create the descriptor, it appeared 'ValueError: index can't contain negative values'. And I used the atoms to create the structure, inclued the Cartesian coordinates I have used in the 'position' for atoms, that really had some negative values. That's the reason for 'ValueError: index can't contain negative values'?

Thanks much for your time!
